From b1b71f5250d24fbf26b38a1e14a02bfcc06a914c Mon Sep 17 00:00:00 2001
From: Artem Borisov <dedsa2002@gmail.com>
Date: Sun, 24 Nov 2019 18:20:56 +0100
Subject: [PATCH] HACK: SF: Force client composition for all layers

For unknown reasons using device composition on all
kirin devices introduces significant image quality drop.
Let's force client composition (no HW overlays) until
we figure the proper solution.

DarkJoker360 edit: Port it to Android 10.

Change-Id: Iccec69e66f27d2d6eff55fc9cb6e5c0e1ef515c5
Signed-off-by: DarkJoker360 <simoespo159@gmail.com>
---
 services/surfaceflinger/SurfaceFlinger.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index ffb7012bb..c6c8e882b 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -1871,10 +1871,10 @@ void SurfaceFlinger::calculateWorkingSet() {
 
             for (auto& layer : display->getOutputLayersOrderedByZ()) {
                 auto& compositionState = layer->editState();
-                compositionState.forceClientComposition = false;
-                if (!compositionState.hwc || mDebugDisableHWC || mDebugRegion) {
+                //compositionState.forceClientComposition = false;
+                //if (!compositionState.hwc || mDebugDisableHWC || mDebugRegion) {
                     compositionState.forceClientComposition = true;
-                }
+                //}
 
                 // The output Z order is set here based on a simple counter.
                 compositionState.z = zOrder++;
